includes/class-wc-gateway-ppec-cart-handler.php

```diff
@@ -362,13 +362,24 @@ public function enqueue_scripts() {
 			wp_enqueue_script( 'wc-gateway-ppec-generate-cart', wc_gateway_ppec()->plugin_url . 'assets/js/wc-gateway-ppec-generate-cart.js', array( 'jquery' ), wc_gateway_ppec()->version, true );
 			wp_localize_script( 'wc-gateway-ppec-generate-cart', 'wc_ppec_generate_cart_context',
 				array(
-					'generate_cart_nonce' => wp_create_nonce( '_wc_ppec_generate_cart_nonce' ),
+					'generate_cart_nonce' => $this->create_nonce( '_wc_ppec_generate_cart_nonce' ),
 					'ajaxurl'             => WC_AJAX::get_endpoint( 'wc_ppec_generate_cart' ),
 				)
 			);
 		}
 	}
 
+	/**
+	 * Ensures a session is active, so that nonce is not invalidated by new session created on AJAX POST request.
+	 */
+	protected function create_nonce( $action ) {
+		if ( ! WC()->session->has_session() ) {
```

Comment:
Do we need to check if `WC()->session` is populated?

Reply:
I don't see it being checked in other places – as far as I understand, it should always be there for front-end requests:
https://github.com/woocommerce/woocommerce/blob/7b56840f360158b3f4a4f4ff924339fd70b82c17/includes/class-woocommerce.php#L476